src/features/web-compat.js

```diff
@@ -635,10 +643,25 @@ export default class WebCompat extends ContentFeature {
                 // Chromium accepts float values for initial-scale
                 const parsedInitialScale = parseFloat(initialScalePart[1])
                 if (parsedInitialScale !== 1) {
-                    this.forceViewportTag(viewportTag, `width=device-width, ${viewportContent}`)
+                    forcedValues.width = 'device-width'
                 }
             }
         }
+
+        const newContent = []
+        'width' in forcedValues && newContent.push(`width=${forcedValues.width}`)
+        'initial-scale' in forcedValues && newContent.push(`initial-scale=${forcedValues['initial-scale']}`)
+        'user-scalable' in forcedValues && newContent.push(`user-scalable=${forcedValues['user-scalable']}`)
+        'maximum-scale' in forcedValues && newContent.push(`maximum-scale=${forcedValues['maximum-scale']}`)
+        if (newContent.length > 0) { // need to override at least one viewport component
+            parsedViewportContent.forEach(([key], idx) => {
+                // @ts-expect-error hasOwn does exist
+                if (!(Object.hasOwn(forcedValues, key))) {
+                    newContent.push(`${viewportContentParts[idx]}`) // reuse the original values, not the parsed ones
```

Comment:
There is a risk that `idx` gets outdated if `viewportContentParts` or `parsedViewportContent` changes item sorting or counts, maybe a comment at their declaration might be helpful?

Reply:
I've marked those variables as readonly, that should address this concern?